transport/transport_test.go

```diff
@@ -830,3 +831,84 @@ func TestIsReservedHeader(t *testing.T) {
 		}
 	}
 }
+
+func TestHTTPToGRPCStatusMapping(t *testing.T) {
+	for k := range httpStatusConvTab {
+		testHTTPToGRPCStatusMapping(t, k)
+	}
+}
+
+func testHTTPToGRPCStatusMapping(t *testing.T, httpStatus int) {
+	lis, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatalf("Failed to listen: %v", err)
+	}
+	done := make(chan error, 1)
+	// Launch an HTTP server to send back header with httpStatus.
+	go func() {
+		var err error
+		defer func() {
+			done <- err
+		}()
+		conn, err := lis.Accept()
+		if err != nil {
+			t.Errorf("Error at server-side while accpeting: %v", err)
+			return
+		}
+		// Read the preface sent by client.
+		if _, err = io.ReadFull(conn, make([]byte, len(http2.ClientPreface))); err != nil {
```

Comment:
We'll have to do the same thing inside the http server handle there, doesn't make it any easier to use that.